web/client/components/widgets/builder/wizard/LegendWizard.jsx

```diff
@@ -51,7 +51,7 @@ module.exports = ({
                     <LegendPreview
                         valid={valid}
                         dependencies={dependencies}
-                        dependenciesMap={data.dependenciesMap}
+                        dependenciesMap={data.dependenciesMap.layers}
```

Comment:
dependenciesMap is used for other things that layers (in general also the viewport of the connected map). So you should continue passing the full dependenciesMap object

Reply:
let me test code with full dependencies.